vignettes/Introduction_To_BiocParallel.Rmd

```diff
@@ -0,0 +1,961 @@
+---
+title: "**Introduction to *BiocParallel***"
+author: "Phyllis Atieno"
+date: "Compiled: 18 October, 2022"
+package: BiocParallel
+vignette: >
+  %\VignetteIndexEntry{Vignette Title}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+output:
+  BiocStyle::html_document
+  
+---
+
+Introduction
+============
+
+Numerous approaches are available for parallel computing in R. The CRAN Task View for
+high performance and parallel computing provides useful high-level summaries and [package
+categorization.](http://cran.r-project.org/web/views/HighPerformanceComputing.html)
+Most Task View packages cite or identify one or more of **snow** , **Rmpi**, **multicore** or **foreach** as
+relevant parallelization infrastructure. Direct support in *R* for *parallel* computing started
+with release 2.14.0 with inclusion of the parallel package which contains modified versions of
+*multicore* and *snow* .
+
+
+A basic objective of *BiocParallel* is to reduce the complexity faced when developing
+and using software that performs parallel computations. With the
+introduction of the `BiocParallelParam` object, *BiocParallel* aims to provide a unified interface to
+existing parallel infrastructure where code can be easily executed in
+different environments. The `BiocParallelParam` specifies the environment of choice as well
+as computing resources and is invoked by 'registration' or passed as an
+argument to the *BiocParallel* functions.
+
+*BiocParallel* offers the following conveniences over the 'roll your own' approach to
+parallel programming.
+
+-   unified interface: `BiocParallelParam` instances define the method of parallel
+    evaluation (multi-core, snow cluster, etc.) and computing resources
+    (number of workers, error handling, cleanup, etc.).
+
+-   parallel iteration over lists, files and vectorized operations: `bplapply`, `bpmapply` and
+    `bpvec` provide parallel list iteration and vectorized operations. `bpiterate` iterates
+    through files distributing chunks to parallel workers.
+
+-   cluster scheduling: When the parallel environment is managed by a
+    cluster scheduler through *batchtools*, job management and result retrieval are
+    considerably simplified.
+
+-   support of `foreach` : The *foreach* and *iterators* packages are fully supported. Registration of
+    the parallel back end uses `BiocParallelParam` instances.
+
+Quick start
+===========
+
+The *BiocParallel* package is available at bioconductor.org and can be downloaded via `BiocManager`:
+
+```
+if (!requireNamespace("BiocManager", quietly = TRUE))
+     install.packages("BiocManager") 
+BiocManager::install("BiocParallel")
+```
+
+Load *BiocParallel*
+
+`library(BiocParallel)`
```

Comment:
This and all (?) code chunks must be evaluated. This is a central feature of vignettes -- the code in the vignette has to actually work.

~~~
```{r}
library(BiocParallel)
```
~~~

Reply:
Hello @mtmorgan , it might take me longer to work on this as I'm still learning R but I'll resolve it as soon as possible. Thank you!